### ISSUE TITLE
API versioning causes "unexpected keyword argument 'version'" at api_root

### ISSUE THREAD

**Issue by bkjones** [django-oscar/django-oscar-api#218]
I have a brand new project using DRF 3.11.0, Oscar 2.0.4, and oscarapi 2.0.2, my urls.py looks like this: 

```from django.apps import apps
from django.contrib import admin
from django.urls import include, path, re_path
from rest_framework_simplejwt.views import TokenObtainPairView

urlpatterns = [
    path('admin/', admin.site.urls),
    path('', include(apps.get_app_config('oscar').urls[0])),
    re_path(r'api/(?P<version>v1/)', include('oscarapi.urls')),
    re_path(r'api/(?P<version>v1)/auth/login/', TokenObtainPairView.as_view(), name='user-login'),
    path('pages/', include('django.contrib.flatpages.urls')),
]

```
I'm using DRF URLPathVersioning. When I hit the API root at `http://localhost:8000/api/v1/` I get an error: 
```
Request Method: GET
Request URL: http://localhost:8000/api/v1/

Django Version: 2.2.13
Python Version: 3.7.6
Exception Type: TypeError at /api/v1/
Exception Value: api_root() got an unexpected keyword argument 'version'

```
However, I am able to hit other URLs that are provided/exposed by oscarapi, for example `/api/v1/products` works. In addition, our own app's DRF views work and they're really simple - just subclassing the generics and they haven't been altered to take a 'version' keyword argument, but don't throw this error. URLs from external apps, like restframework_simplejwt which we use for authentication, also work with no updates from us. 

Can someone help me understand what's happening?

**Comment by maerteijn** [comment]
The `api_root` view is not a class based view but a really simple view function:
```python
@api_view(("GET",))
def api_root(request, format=None):
   ...
```
So it won't take any parameters other than `format`. OscarAPI was not intented to be developed to have multiple version support in the same codebase so it was never tested with the [version schemes ](https://www.django-rest-framework.org/api-guide/versioning/#versioning) available in Django Rest Framework.

I guess that just adding:
```python
def api_root(request, format=None, **kwargs)
``` 

to the view would let it accept the `version` keyword argument and that you can do whatever you like with the `version` parameter in the views you [can override](https://django-oscar-api.readthedocs.io/en/latest/usage/customizing_oscarapi.html).

I'll check this out the following week to see what impact this has. Keep you posted.

**Comment by bkjones** [comment]
Thanks, I appreciate you checking it out! 

My understanding is that the `api_view` decorator wraps the decorated function in a class that subclasses DRF's APIView. That being the case, it can be treated like a class, so, I'm trying to override it. OscarAPI sees, loads, and executes my override (yay!) but I'm still struggling a bit (this is my first time doing anything with oscarapi). This would definitely be a lot easier with the change you suggest (though of course I have zero knowledge of other implications of that change in the oscarapi codebase). 

Here's my override code for the api_root: 

```
from oscarapi.views.root import api_root as oscar_api_root
from rest_framework.decorators import api_view
import logging

logger = logging.getLogger(__name__)
logger.critical("OVERRIDE VIEW MODULE LOADED!")
@api_view()
def api_root(request, version=None):
    logger.critical("OVERRIDE VIEW CALLED!")
    print("VERSION {}".format(version))
    return oscar_api_root(request)
```

Only trouble now is that I get an error that now says:
`The `request` argument must be an instance of `django.http.HttpRequest`, not `rest_framework.request.Request`.`

This is a bit of a surprise being that this is built on DRF, but I'm sure there's some valid reason for this that I don't understand. Maybe there's some way to turn a DRF Request into a Django Request object. I'll keep trying. Thanks again for your reply and your work on this project.

**Comment by maerteijn** [comment]
You should not add the `@api_view()` decorator again to your overriden view. Then you can call the original `oscar_api_root` without any problems.

But that will only 'fix' this particular view, and there are more in OscarAPI which don't accept any extra arguments.

I have a structural [fix](https://github.com/django-oscar/django-oscar-api/commit/88760e3a5ce49b5e47ddb1e6f328e59ee5519f71l) waiting in another branch. Probably will be merged next week.

**Comment by maerteijn** [comment]
The fix is merged now, so it will be there in the next release.

**Comment by maerteijn** [comment]
Released in version 2.1.0
